src/api/app/controllers/source_controller.rb

```diff
@@ -1560,11 +1560,11 @@ def package_command_linktobranch
 
     # convert link to branch
     rev = ''
-    if not pkg_rev.nil? and not pkg_rev.empty?
+    if !pkg_rev.nil? && !pkg_rev.empty?
```

Comment:
@Ana06 when use said pkg is " " I guess that you are talking about pkg_rev....

If pkg_rev  is " " then maybe you need to strip it on line 1558 (and also 1559) before.
```ruby
pkg_rev = params[:rev].strip
pkg_linkrev = params[:linkrev].strip
```